File: ASP_Api/ec_project/Models/Coupon.cs
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ec_project.Models
{
    public class Coupon
    {
        [BsonId]
        [BsonRepresentation(BsonType.ObjectId)]
        public string _id { get; set; }
        public string name { get; set; }
        public decimal value { get; set; }
        public int quantity { get; set; }
        public int use_quantity { get; set; }
        public ulong exp { get; set; }
        public ulong create_time { get; set; }
        public ulong edit_time { get; set; }
        public Admin[] modifier { get; set; }
        public string note { get; set; }
        public string status { get; set; }
    }
}

File: Angular/src/app/Data/Types/Coupon.ts
import { Admin } from "./Admin";

export class Coupon{
    _id:string;
    name:string;
    value:number;
    quantity:number;
    use_quantity: number;
    exp: number;
    create_time: number;
    edit_time: number;
    modifier: Admin[];
    note: string;
    status: string;

    constructor(object: any){
        this._id=object._id;
        this.name=object.name;
        this.value=object.value;
        this.quantity=object.quantity;
        this.use_quantity=object.use_quantity;
        this.exp=object.exp;
        this.create_time=object.create_time;
        this.edit_time=object.edit_time;
        this.modifier=object.modifier;
        this.note=object.note;
        this.status=object.status;
    }
}